### PR TITLE
PID-controlled cursor with tuner

### PR DESCRIPTION
**Pull request by Clstroud**
* Integrates a tweaked version of the (awesome) [Pulse](https://github.com/cieslakdawid/Pulse) library for better cursor smoothing. Eventually we can probably drop the dependency and implement the core PID formula itself, but for now it's nice to have the whole thing since it comes with a tuning UI 
* Makes it so that touches disable gaze tracking for 5 seconds so the two interaction schemes don't conflict

### CODE REVIEW

File: EyeSpeak/Features/Presets/PresetsViewController.swift [hunk 1]
```diff
@@ -285,21 +296,34 @@ class PresetsViewController: UICollectionViewController {
...
+                
+                // TODO: discuss with design if we want to cache the user's currently-entered text instead
+                // of just clearing it
+                currentSpeechText = showKeyboard ? HintText.keyboard.rawValue : HintText.preset.rawValue
             }
         case .presetItem(let text):
             currentSpeechText = text
-            AVSpeechSynthesizer.shared.speak(currentSpeechText)
+            // Dispatch to get off the main queue for performance
+            DispatchQueue.global(qos: .userInitiated).async {
```
Comment (jrtb) [inline EyeSpeak/Features/Presets/PresetsViewController.swift]: nice!

File: EyeSpeak/Features/Presets/PresetsViewController.swift [hunk 2]
```diff
@@ -285,21 +296,34 @@ class PresetsViewController: UICollectionViewController {
...
+                AVSpeechSynthesizer.shared.speak(text)
+            }
         case .category(let category):
             selectedCategory = category
             return
         case .keyboardFunctionButton(let functionType):
             switch functionType {
             case .space:
-                currentSpeechText.append(" ")
+                self.didSelectCharacter(" ")
```
Comment (jrtb) [inline EyeSpeak/Features/Presets/PresetsViewController.swift]: lose self. ?

File: EyeSpeak/HeadGazeLib/HeadGazeTrackingInterpolator.swift [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+    final func update(withFrame frame: ARFrame, correctionAmount: CGSize, scale: CGFloat) {
+
+        guard let faceAnchor = faceAnchor, let view = view else { return }
+
+        let blendShapes = faceAnchor.blendShapes
+        let lBlinkAmount = blendShapes[.eyeBlinkLeft]?.floatValue ?? 0.0
+        let rBlinkAmount = blendShapes[.eyeBlinkRight]?.floatValue ?? 0.0
+
+        let blinkThreshold: Float = 0.3 // 0.05
```
Comment (jrtb) [inline EyeSpeak/HeadGazeLib/HeadGazeTrackingInterpolator.swift]: ok to remove the comment here?

File: EyeSpeak/HeadGazeLib/HeadGazeTrackingInterpolator.swift [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            // that a face is actually trackable. This is to prevent
+            // the cursor jumping wildly when first (re-)entering the view
+            needsResetOnNextUpdate = true
+            previousGaze = nil
+            event = nil
+        }
+    }
+
+    /**
+     return head gaze projection in 2D NDC coordinate system
```
Comment (jrtb) [inline EyeSpeak/HeadGazeLib/HeadGazeTrackingInterpolator.swift]: ❤️ this comment!

File: EyeSpeak/HeadGazeLib/Interpolation/PIDInterpolator.swift [hunk 5]
```diff
@@ -0,0 +1,71 @@
+//
+//  PIDInterpolator.swift
+//  EyeSpeak
+//
+//  Created by Chris Stroud on 2/12/20.
+//  Copyright © 2020 WillowTree. All rights reserved.
+//
+
```
Comment (jrtb) [inline EyeSpeak/HeadGazeLib/Interpolation/PIDInterpolator.swift]: Summary comment on purpose of the class?

File: EyeSpeak/HeadGazeLib/UIHeadGazeViewController.swift [hunk 6]
```diff
@@ -62,8 +53,12 @@ class UIHeadGazeViewController: UIViewController, ARSessionDelegate, ARSCNViewDe
         sceneview?.delegate = self
         sceneview?.session.delegate = self
         sceneview?.isHidden = true
-
+        sceneview?.preferredFramesPerSecond =  UIScreen.main.maximumFramesPerSecond //8
```
Comment (jrtb) [inline EyeSpeak/HeadGazeLib/UIHeadGazeViewController.swift]: still need the comment?

File: EyeSpeak/Features/Presets/PresetsViewController.swift (hunk 2 above)
Comment (pg8wood) [inline EyeSpeak/Features/Presets/PresetsViewController.swift]: @jrtb I gotchu

```suggestion
                didSelectCharacter(" ")
```

**Pull request merged**